src/main/java/org/commcare/xml/GlobalParser.java

```diff
@@ -0,0 +1,35 @@
+package org.commcare.xml;
+
+import org.commcare.suite.model.GeoOverlay;
+import org.commcare.suite.model.Global;
+import org.javarosa.xml.ElementParser;
+import org.javarosa.xml.util.InvalidStructureException;
+import org.kxml2.io.KXmlParser;
+import org.xmlpull.v1.XmlPullParserException;
+
+import java.io.IOException;
+import java.util.Vector;
+
+/**
+ * Parser used in DetailParser to parse the defintion of Global element used in case-select and case-detail views
+ */
+class GlobalParser extends ElementParser<Global> {
+
+    static final String NAME_GLOBAL = "global";
+
+    GlobalParser(KXmlParser parser) {
+        super(parser);
+    }
+
+    @Override
+    public Global parse() throws InvalidStructureException, IOException, XmlPullParserException {
+        Vector<GeoOverlay> geoOverlays = new Vector<>();
```

Comment:
How does this parse go when the global block is empty? It might be easiest for HQ to generate a bunch of empty `<global/>` blocks rather than only inject them when needed, so if we can plan around that it's really helpful.

Reply:
In case of an empty global tag, `Global` will just get initialized with an empty  `geoOverlays` list.